src/pages/workspaces/workspace/useWorkspace.ts

```diff
@@ -45,9 +46,10 @@ export const useWorkspace = (namespace, name) : WorkspaceDetails => {
   const accessNotificationId = useRef()
   const cachedWorkspace = useStore(workspaceStore)
   const workspace = cachedWorkspace && _.isEqual({ namespace, name }, _.pick(['namespace', 'name'], cachedWorkspace.workspace)) ? cachedWorkspace : undefined
-  const [{ location, locationType }, setGoogleStorage] = useState({
-    location: defaultLocation, locationType: locationTypes.default // These default types are historical
-  })
+  const [{ location, locationType, fetchedLocation }, setGoogleStorage] =
+    useState<{ fetchedLocation: 'SUCCESS' | 'ERROR' | undefined; location: string; locationType: string }>({
+      fetchedLocation: undefined, location: defaultLocation, locationType: locationTypes.default // These default types are historical
+    })
```

Comment:
I experimented with using `LoadedState` and `useLoadedData`, but I couldn't get it working properly, and it certainly wasn't seeming any simpler/clearer.

Reply:
I don't think it's a simple drop-in here: this whole thing looks like it badly needs a state management solution.